processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -643,16 +661,14 @@ func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) PodContain
 	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag {
 		for _, spec := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
 			container := &Container{}
-			nameTagSep := strings.LastIndex(spec.Image, ":")
-			if c.Rules.ContainerImageName {
-				if nameTagSep > 0 {
-					container.ImageName = spec.Image[:nameTagSep]
-				} else {
-					container.ImageName = spec.Image
+			name, tag, err := parseNameAndTagFromImage(spec.Image)
```

Comment:
Could we reflect this logic in the documentation as well? Something like what was described at https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36131#issuecomment-2452475338

Reply:
done